evap/student/templates/student_vote.html

```diff
@@ -194,12 +197,52 @@ <h3 class="mb-3">{{ evaluation.full_name }} ({{ evaluation.course.semester.name
                     if (textResultsPublishConfirmation.bottomCard) {
                         updateTextResultsPublishConfirmation();
                     }
-                }
+                },
+                onSave: function () {
+                    const lastSaved = $('#last-saved');
+                    const timeNow = new Date();
+                    localStorage.setItem(lastSavedStorageKey, timeNow.toString());
+                },
             });
+
+            function updateLastSavedLabel() {
+                    const timeNow = new Date();
+                    const lastSavedLabel = $('#last-saved');
+                    const lastSavedDate = new Date(localStorage.getItem(lastSavedStorageKey))
+                    const delta = Math.round((timeNow - lastSavedDate) / 1000)
+                    const languageCode = "{{ LANGUAGE_CODE }}"
+                    const relativeTimeFormat = new Intl.RelativeTimeFormat(languageCode)
+                    let timeStamp = "{% trans 'unknown' %}"
+                    if (delta < 3) {
+                        timeStamp = "{% trans 'just now' %}";
+                    } else if (delta < 60) {
+                        timeStamp = relativeTimeFormat.format(-delta, 'seconds');
+                    } else if (delta < 60 * 30) {
+                        timeStamp = relativeTimeFormat.format(-Math.round(delta / 60), 'minutes');
+                    } else if (delta < 60 * 60 * 12) {
+                        timeStamp = padWithLeadingZeros(lastSavedDate.getHours()) + ":" + padWithLeadingZeros(lastSavedDate.getMinutes());
+                    } else {
```

Comment:
I’d check here that there is a sensible `delta` here. If – for any reason – the local storage item is empty, this code will print out 1970-01-01 01:00 as saved date. (Or for the value `""`: NaN-NaN-NaN NaN:NaN).

Reply:
NaN-NaN-NaN NaN:NaN Batman!

https://www.destroyallsoftware.com/talks/wat

Ok I'll show myself out.

Reply:
Should be resolved with newest commit